tests/mode.test.ts

```diff
@@ -0,0 +1,87 @@
+import fs from 'fs';
+import { join, basename, extname } from 'path';
+import CodeMirror from 'codemirror';
+import 'codemirror/mode/meta';
+import 'codemirror/addon/mode/simple';
+import { defineRobotMode, MODE_NAME } from '../src/mode';
+
+const fixtureDir = join(__dirname, `${basename(__filename, '.ts')}.fixtures`);
+const fixtures = fs
+  .readdirSync(fixtureDir)
+  .filter((x) => extname(x).toLowerCase() === '.robot')
+  .map((x) => [basename(x, '.robot')]);
+
+// https://github.com/jsdom/jsdom/issues/3002
+document.createRange = () => {
+  const range = new Range();
+  range.getBoundingClientRect = jest.fn();
+  range.getClientRects = () => {
+    return {
+      item: () => null,
+      length: 0,
+      [Symbol.iterator]: jest.fn()
+    };
+  };
+  return range;
+}
+
+const tokenTypeMap: { [key: string]: string } = {
+  'cm-header': 'header',
+  'cm-atom': 'atom',
+  'cm-keyword': 'keyword',
+  'cm-attribute': 'attr',
+  'cm-operator': 'operator',
+  'cm-variable-2': 'var2',
+  'cm-string': 'string',
+}
+
+const mapToTokenTypes = (value: string): string => {
+  const tokenTypes = value?.split(' ').map((x) => {
+    const tokenType = tokenTypeMap[x];
+    return tokenType ? tokenType : x;
+  }).join('|');
+  return tokenTypes ? `<${tokenTypes}>` : null;
+}
+
+const parseTokens = (
+  editor: CodeMirror.EditorFromTextArea,
+  body: HTMLElement,
+): string[] => {
+  const lineCount = editor.lineCount();
+  const lines: string[] = [];
+  for (let line = 0; line < lineCount; line++) {
+    editor.setCursor({line: line, ch: 0});
+    const nodes = body.querySelector('.CodeMirror-scroll .CodeMirror-line span').childNodes;
+    const spans: any[] = [];
+    nodes.forEach((x) => spans.push(x));
+    const allTokens = spans.map((span) => ({
+      types: mapToTokenTypes(span.attributes?.class?.value),
+      text: span.innerHTML,
+    }));
+    const tokens = allTokens
+      .filter((x) => x.text !== undefined && x.text !== '&nbsp;')
+      .map((token) => `${token.types}: ${token.text}`);
+    lines.push([`#${line}`, ...tokens].join(' '));
+  }
+  return lines;
+}
+
+defineRobotMode({ CodeMirror } as any);
+document.body.innerHTML = `<textarea id='code' />`
+const codeElement = document.getElementById('code') as HTMLTextAreaElement;
+const editor = CodeMirror.fromTextArea(codeElement, { mode: MODE_NAME });
+
+test('fixture count', () => expect(fixtures.length).toBe(1));
```

Comment:
maybe just, "more than one"? It seems like having to update this per PR would get finicky.